settings.gradle

```diff
@@ -192,3 +194,27 @@ if (JavaVersion.current() == JavaVersion.VERSION_1_8) {
   }
 }
 
+dependencyResolutionManagement {
+  versionCatalogs {
+    libs {
+      from(files("./libs.versions.toml"))
+
+      library("delta-standalone", "io.delta:delta-standalone_${scalaVersion}:0.6.0")
+      library("flink115-table-planner", "org.apache.flink:flink-table-planner_${scalaVersion}:1.15.0")
+      library("flink116-table-planner", "org.apache.flink:flink-table-planner_${scalaVersion}:1.16.2")
+      library("flink117-table-planner", "org.apache.flink:flink-table-planner_${scalaVersion}:1.17.1")
+      library("jackson211-module-scala", "com.fasterxml.jackson.module:jackson-module-scala_${scalaVersion}:2.11.4")
+      library("jackson212-module-scala", "com.fasterxml.jackson.module:jackson-module-scala_${scalaVersion}:2.12.3")
+      library("jackson213-module-scala", "com.fasterxml.jackson.module:jackson-module-scala_${scalaVersion}:2.13.4")
+      library("scala-collection-compat", "org.scala-lang.modules:scala-collection-compat_${scalaVersion}:2.6.0")
```

Comment:
For projects that do not publish a BOM, it's a bit trickier to remove the hard-coded versions here.
You probably have to declare these dependencies in `libs.versions.tom` as well, but replace the hard-coded versions here with something like this in the `build.gradle` files:
```groovy
implementation("org.scala-lang.modules:scala-collection-compat_${scalaVersion}:${libs.versions.scalaCollectionCompatVersion}")
```

And in the `libs.versions.toml` file declare the libararies for one Scala version (or all, up to you). That way, those versions are managed via the version catalog as well.

Reply:
> That way, those versions are managed via the version catalog as well.

It's managed by a version catalog, it's just not part of toml file but a part of version catalog declaration in settings.gradle. 


> implementation("org.scala-lang.modules:scala-collection-compat_${scalaVersion}:${libs.versions.scalaCollectionCompatVersion}")

I tried this construct on delta lake dependency and I get errors from Maven while it tries to resolve version that is supposed to come from toml file. Any ideas? Do I need to do something else?

I added 

```
[versions]
deltaVersion = "0.6.0"
...
[libraries]
delta-standalone = { module = "io.delta:delta-standalone_2.12", version.ref = "deltaVersion" }
```

and then I tried to use it like so:

```
    compileOnly "io.delta:delta-standalone_${scalaVersion}:${libs.versions.deltaVersion}"
```

I get:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':iceberg-delta-lake:compileJava'.
> Could not resolve all files for configuration ':iceberg-delta-lake:compileClasspath'.
   > Could not resolve io.delta:delta-standalone_2.12:provider(?).
     Required by:
         project :iceberg-delta-lake
      > Could not resolve io.delta:delta-standalone_2.12:provider(?).
         > Could not get resource 'https://artifacts.apple.com/libs-release/io/delta/delta-standalone_2.12/provider(%3F)/delta-standalone_2.12-provider(%3F).pom'.
            > Could not GET 'https://artifacts.apple.com/libs-release/io/delta/delta-standalone_2.12/provider(%3F)/delta-standalone_2.12-provider(%3F).pom'.
               > artifacts.apple.com
      > Could not resolve io.delta:delta-standalone_2.12:provider(?).
         > Could not get resource 'https://artifacts.apple.com/libs-snapshot/io/delta/delta-standalone_2.12/provider(%3F)/delta-standalone_2.12-provider(%3F).pom'.
            > Could not GET 'https://artifacts.apple.com/libs-snapshot/io/delta/delta-standalone_2.12/provider(%3F)/delta-standalone_2.12-provider(%3F).pom'.
               > artifacts.apple.com
```

Reply:
Maybe add a `.get()` ??